### ISSUE TITLE
Incorrect parenthesization of arrow functions

### ISSUE THREAD

**Issue by Conduitry** [Rich-Harris/code-red#31]
```js
cr.print(cr.x`foo || (bar => bar)`).code
```

results in

```js
foo || bar => bar
```

which is syntactically incorrect. The arrow function needs to be wrapped in parentheses. I'm not sure what precedence rules are in play here or what needs changing.

Ref https://github.com/sveltejs/svelte/issues/4384

**Comment by Conduitry** [comment]
Aside:

```js
cr.print(cr.x`foo || bar => bar`).code
```

results in

```js
foo || bar
```

for some reason. I guess Acorn silently stops parsing when it sees the invalid `=>`? I'm not sure whether that should be considered a bug.

**Comment by Rich-Harris** [comment]
Yeah, Acorn just ignores anything past the first valid expression. Definitely a bug for our situation

**Comment by Rich-Harris** [comment]
fixed in 0.1.1
